src/vox_scene_info.rs

```diff
@@ -0,0 +1,213 @@
+use bevy::{
+    prelude::{
+        AssetServer, Assets, BuildChildren, Commands, Component, ComputedVisibility, Entity,
+        GlobalTransform, Handle, Mesh, Name, PbrBundle, StandardMaterial, Transform, Visibility,
+    },
+    reflect::{TypePath, TypeUuid},
+    utils::HashMap,
+};
+use dot_vox::{Layer, SceneNode};
+
+#[derive(Debug, TypeUuid, TypePath, Clone)]
+#[uuid = "39cadc56-aa9c-4543-8640-a018b74b5052"]
+
+pub struct VoxSceneInfo {
+    pub scenes: Vec<SceneNode>,
+    pub layers: Vec<Layer>,
+    // layers hidden status
+    pub layer_map: HashMap<u32, bool>,
+}
+
+#[derive(Debug, Clone, Component)]
+pub struct LayerData(pub u32);
+
+impl VoxSceneInfo {
+    pub fn new(scenes: Vec<SceneNode>, layers: Vec<Layer>) -> Self {
+        Self {
+            scenes: scenes,
+            layers: layers.clone(),
+            layer_map: layers
+                .clone()
+                .into_iter()
+                .map(|x| {
+                    let number = x
+                        .attributes
+                        .get("_name")
+                        .map(|s| s.parse().unwrap())
+                        .unwrap_or(u32::MAX);
+                    let hidden = x
+                        .attributes
+                        .get("_hidden")
+                        .map(|s| s == "1")
+                        .unwrap_or(false);
+                    (number, hidden)
+                })
+                .collect(),
+        }
+    }
+
+    pub fn all_loaded(
```

Comment:
Current function name doesn't explicitly tell much about what it does, documentation would be appreciated.

```suggestion
    pub fn assert_all_loaded(
```